addons/project/models/project_project.py

```diff
@@ -208,11 +208,26 @@ def _read_group_stage_ids(self, stages, domain, order):
     milestone_count = fields.Integer(compute='_compute_milestone_count', groups='project.group_project_milestone')
     milestone_count_reached = fields.Integer(compute='_compute_milestone_reached_count', groups='project.group_project_milestone')
     is_milestone_exceeded = fields.Boolean(compute="_compute_is_milestone_exceeded", search='_search_is_milestone_exceeded')
+    milestone_progress = fields.Integer("Milestones Reached", compute='_compute_milestone_reached_count')
+    next_milestone_id = fields.Many2one('project.milestone', compute='_compute_next_milestone_id', store=True)
+    is_next_milesotone_unreached_and_late = fields.Boolean(compute='_compute_is_next_milesotone_unreached_and_late')
 
     _sql_constraints = [
         ('project_date_greater', 'check(date >= date_start)', "The project's start date must be before its end date.")
     ]
 
+    @api.depends('milestone_ids', 'milestone_ids.is_reached', 'milestone_ids.deadline')
+    def _compute_next_milestone_id(self):
+        for project in self:
+            next_milestones = project.milestone_ids.filtered(lambda m: not m.is_reached)
+            project.next_milestone_id = next_milestones and next_milestones[0].id
+
+    @api.depends('next_milestone_id.is_reached', 'next_milestone_id.deadline')
+    def _compute_is_next_milesotone_unreached_and_late(self):
+        for project in self:
+            next_milestone = project.next_milestone_id
+            project.is_next_milesotone_unreached_and_late = not next_milestone.is_reached and next_milestone.deadline and next_milestone.deadline < fields.Date.today()
```

Comment:
```suggestion
        today = fields.Date.today()
        for project in self:
            next_milestone = project.next_milestone_id
            project.is_next_milesotone_unreached_and_late =\
                not next_milestone.is_reached and\
                next_milestone.deadline and\
                next_milestone.deadline < today
```
- Better to break the line imo
- Get `fields.Date.today()` outside the loop